scripts/barclamp_mgmt_lib.rb

```diff
@@ -421,24 +422,36 @@ def bc_install_layout_1_app(from_rpm, bc_path)
   debug "Component #{component} added to Crowbar Framework.  Review #{filelist} for files created."
 end
 
-# sync the chef parts
-def bc_install_layout_1_chef
+def run_rake_task(task_name, log)
+  crowbar_user = Etc.getpwnam("crowbar")
   rails_env = ENV["RAILS_ENV"] || "production"
-  upload_cmd = "cd #{CROWBAR_PATH} && RAILS_ENV=#{rails_env} bin/rake chef:upload:all"
-  system "su -s /bin/sh - crowbar sh -c \"#{upload_cmd}\""
+  debug("spawning bin/rake #{task_name}")
+  rake_process = spawn({"RAILS_ENV" => rails_env},
+                       "bin/rake --silent #{task_name}",
+                       :uid => crowbar_user.uid,
```

Comment:
We can't change the uid or gid from a user. The script is not always executed as root but crowbar is allowed to use su. If the script is executed as crowbar user then setting the uid or gid will fail.